node/router/src/helpers/sync.rs

```diff
@@ -67,8 +68,9 @@ impl Hash for PeerPair {
 /// - When a request is timed out, the `requests`, `request_timestamps`, and `responses` map remove the entry for the request height;
 #[derive(Clone, Debug)]
 pub struct Sync<N: Network> {
-    /// The listener IP of this node.
-    local_ip: SocketAddr,
+    /// The TCP stack of this node.
```

Comment:
Can you help explain the rationale behind the code changes in this file? I'm not following why this would make testing bettter.

My preference would be for the `Sync` module to be a subchild of `Router` and adhere strictly to `Router`'s state instead of carrying on `Tcp` state so that we ensure network abstracts remain pure to `Router`.